terraform/README.md

```diff
@@ -0,0 +1,168 @@
+
+**The infrastructure used to run this application can be categorized into two distinct types: _ephemeral_ and _persistent_**
+
+* **Ephemeral infrastructure** is all the infrastructure that is recreated each time the application deploys. Ephemeral infrastructure includes the "application(s)" (as defined in Cloud.gov), the EC2 instances the application runs on, and the routes that application utilizes. Our CircleCI configuration describes this infrastructure and deploys it to Cloud.gov.
+* **Persistent infrastructure** is all the infrastructure that remains constant and unchanged despite application deployments. Persistent infrastructure includes the database used in each development environment. Our Terraform configuration files describe this infrastructure and instantiates it on Cloud.gov.
+
+> This concept is often referred to as _mutable_ vs _immutable_ infrastructure.
+
+### Infrastructure as Code
+
+A high-level configuration syntax, called [Terraform language][language], describes our infrastructure. This allows a blueprint of our system to be versioned and treated as we do any other code. This configuration can be acted on locally by a developer if deployments need to be created manually, but it is mostly and ideally executed by CircleCI.
+
+###  Terraform workflow
+
+Terraform integrates into our CircleCI pipeline via the [Terraform orb][orb], and is formally described in our `deploy-infrastructure` CircleCI job. Upon validating its configuration, Terraform reads the current state of any already-existing remote objects to make sure that the Terraform state is up-to-date, and compares the current configuration to the prior state, noting all differences. Terraform creates a "plan" and proposes a set of change actions that should, if applied, make the remote objects match the configuration – this is the essence of _infrastructure as code_.
+
+### Terraform state
+
+We use an S3 bucket created by Cloud Foundry in Cloud.gov as our remote backend for Terraform. This backend maintains the "state" of Terraform and makes it possible for us to make automated deployments based on changes to the Terraform configuration files. **This is the only part of our infrastructure that must be manually configured.**
+
+Note that a single S3 bucket maintains the Terraform State for both the development and staging environments, and this instance is deployed in the development space.
+
+|   | development  | staging  | production | 
+|---|---|---|---|
+| S3 Key | `terraform.tfstate.dev`   | `terraform.tfstate.staging`  | `terraform.tfstate.prod`  |
+| Service Space | `tanf-dev`  | `tanf-dev`  | `tanf-prod`  |
+
+
+## Local Set Up For Manual Deployments
+
+Sometimes a developer will need to run Terraform locally to perform manual operations. Perhaps a new TF State S3 bucket needs to be created in another environment, or there are new services or other major configuration changes that need to be tested first.
+
```

Comment:
@jorgegonzalez can we add the install steps for jq and circleci (in case anyone needs it)?

Reply:
Thanks, added!